apps/web/pages/api/cron/monthlyDigestEmail.ts

```diff
@@ -0,0 +1,339 @@
+import type { Prisma } from "@prisma/client";
+import type { NextApiRequest, NextApiResponse } from "next";
+import { z } from "zod";
+
+import dayjs from "@calcom/dayjs";
+import { send30DayDigestEmails } from "@calcom/emails/email-manager";
+import { EventsInsights } from "@calcom/features/insights/server/events";
+import { getTranslation } from "@calcom/lib/server";
+import prisma from "@calcom/prisma";
+
+const querySchema = z.object({
+  page: z.coerce.number().min(0).optional().default(0),
+});
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  const apiKey = req.headers.authorization || req.query.apiKey;
+
+  if (process.env.CRON_API_KEY !== apiKey) {
+    res.status(401).json({ message: "Not authenticated" });
+    return;
+  }
+  if (req.method !== "POST") {
+    res.status(405).json({ message: "Invalid method" });
+    return;
+  }
+
+  const delay = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+  const pageSize = 90; // Adjust this value based on the total number of teams and the available processing time
+  let { page: pageNumber } = querySchema.parse(req.query);
+
+  const date30DaysAgo = new Date(new Date().setDate(new Date().getDate() - 30));
+
+  while (true) {
+    const teams = await prisma.team.findMany({
+      where: {
+        slug: {
+          not: null,
+        },
+        createdAt: {
+          // created before last 30days
+          lte: date30DaysAgo,
+        },
+      },
+      select: {
+        id: true,
+        createdAt: true,
+        members: true,
+        name: true,
+      },
+      skip: pageNumber * pageSize,
+      take: pageSize,
+    });
+
+    if (teams.length === 0) {
+      break;
+    }
+
+    for (const team of teams) {
+      const createdAt = team.createdAt;
+
+      // Setting both times to midnight as we will only compare the days
+      date30DaysAgo.setHours(0, 0, 0, 0);
+      createdAt.setHours(0, 0, 0, 0);
+
+      // Calculate the difference in milliseconds between createdAt and date30DaysAgo
+      const timeDifference = date30DaysAgo.getTime() - createdAt.getTime();
+
+      // Calculate the difference in days
+      const daysDifference = Math.floor(timeDifference / (1000 * 60 * 60 * 24));
+
+      // If the daysDifference is a multiple of 30
+      if (daysDifference % 30 === 0) {
```

Comment:
Our goal is to trigger this email every 30 days for every team, so here we're first setting the time for the createdAt date so that there are no conflicts with time and we only compare days, next we're calculating the difference between the createdAt date and the date 30 days ago if the difference is a multiple of 30 that means another month has passed for that team and thus we can trigger an email.

Reply:
can we do last day of a month instead?

Reply:
Just to understand you clearly you're suggesting that we trigger this email at the end of every month for every team and the team receives the updates for only that particular month right? 

So in the case of a team created just 10 days before the end of a month will we provide them the updates or these updates would just be for teams created at least before or at the first of the month?

Can be done just need to be sure about the logical requirements.

Reply:
@PeerRich